packages/core/src/Page/index.js

```diff
@@ -432,6 +432,12 @@ class Page {
       this.collectNavigableHeadings(content);
       const pageNavTitleHtml = this.generatePageNavTitleHtml();
       const pageNavHeadingHTML = this.generatePageNavHeadingHtml();
+
+      if ((!pageNavTitleHtml || pageNavTitleHtml.length === 0)
+        && (!pageNavHeadingHTML || pageNavHeadingHTML.length === 0)) {
```

Comment:
my understanding with #2054 is that the fix there is still ideal (dont render the data-bs attribute if we dont need it since we dont know whta might happen), even if this fixes it as a side effect as well.

---

the second .length checks here are unnecessary, empty strings are falsy.

Reply:
@domlimm yup as mentioned above, my PR should still stand as it tackles the bs attribute part. Your PR should be targeting just the #2055 part

Reply:
thanks for highlighting that! will make the change now

Reply:
@ang-zeyu @tlylt also, thanks so much for the help guys!

Reply:
> Proposed commit message: (wrap lines at 72 characters)

Let's add a message for the squashed commit

Reply:
Will do!